### DISCUSSION TITLE
Pool Heater tile does not show up. Would also like to remove Spa Water Temperature.

### DISCUSSION THREAD

**Discussion by Bugman1400** [aqualinkd/AqualinkD#328] [General]
Not sure why the Pool Heater tile does not show, but the Spa Water Temperature does. I don't have a Spa (wish I did, but I don't).

`   // Display order of devices.  Tiles will be displayed in the order below,
    // any devices you don't want to see you can comment the ID. (with // e.g. `//"Solar_Heater",` )
    // If the device isn't listed below is will NOT be shown.
    // For a complete list returned from your particular aqualinkd instance
    // use the below URL and look at the ID value for each device.
    // http://aqualink.ip.address/?command=devices
    var devices = [
        "Filter_Pump",
//        "Spa_Mode",
        "Aux_1",
        "Aux_2",
        "Aux_3",
//        "Aux_4",
//        "Aux_5",
//        "Aux_6",
//        "Aux_7",
//        "Aux_B1",
//        "Aux_B2",
//        "Aux_B3",
//        "Aux_B4",
//        "Aux_B5",
//        "Aux_B6",
//        "Aux_B7",
//        "Aux_B8",
        "Pool_Heater",
//        "Spa_Heater",
//        "SWG",
        //"SWG/Percent",
//        "SWG/PPM", 
        //"SWG/Boost",
        "Temperature/Air",
        "Temperature/Pool",
//        "Temperature/Spa",
        "Pool_Water",
//        "Spa_Water",
        "Freeze_Protect",
//        "CHEM/pH",
//        "CHEM/ORP",
//        "Solar_Heater",
      ];
  
      // This get's picked up by dynamic_config.js and used as mode 0
      var light_program = [
        "Voodoo Lounge - Show",
        "Blue Sea",
        "Royal Blue",
        "Afternoon Skies",
        "Aqua Green",
        "Emerald",
        "Cloud White",
        "Warm Red",
        "Flamingo",
        "Vivid Violet",
        "Sangria",
        "Twilight - Show",
        "Tranquility - Show",
        "Gemstone - Show",
        "USA - Show",
        "Mardi Gras - Show",`

![Screenshot from 2024-08-10 18-02-01](https://github.com/user-attachments/assets/d6f4e45f-3499-4c0c-8728-9119a140fa7c)

**Comment by sfeakes** [comment]
You have the configuration incorrect, that is showing you have configured for a Combo panel, ie pool & spa.  `/etc/aqualinkd.conf` is the file you need to modify.  

Read the below link.  and the `panel_type` section, looks like you have `RS-4 Combo` when you want `RS-4 Only`. Also make sure to adjust the button section correctly.
https://github.com/sfeakes/AqualinkD/wiki#Configuration

`serial_logger` should give you the config option to start with.

**Comment by Bugman1400** [comment]
These are my serial logging results......
![Screenshot from 2024-08-11 08-03-50](https://github.com/user-attachments/assets/ff2839e9-fdc6-41be-8c58-85f208f6b951)

I changed my aqualinkd.conf to:
panel_type = RS-4 Only (I had this at RS-4 Combo initially)
device_id=0x08
rssa_device_id=0x48
extended_device_id=0x30
extended_device_id_programming = yes
read_RS485_vsfPump = yes
button_01_label=Filter Pump

button_02_label=Waterfall Pump
button_02_pumpID=0x10
button_02_pumpIndex=1

button_03_label=Pool Heater (Initially had button_10_label=Pool Heater)

button_04_label=Pool Light
button_04_lightmode=2

button_05_label=NONE

button_06_label=NONE
#button_05_lightMode=0

button_07_label=NONE
#button_05_lightMode=0

button_08_label=NONE

button_09_label=NONE

button_10_label=NONE

button_11_label=NONE

button_12_label=NONE

Now, I get this..................
![Screenshot from 2024-08-11 08-12-38](https://github.com/user-attachments/assets/25839b06-ba4f-4c20-8949-0b1fef982b54)

**Reply by sfeakes** [reply]
So you have a dual panel, but no dual equipment.  In that case, you should set it back to `RS-4 Combo` as the panels act differently and have different button configs.

You have not configured the pool heater on the buttons part of the config, that's why it's not showing up.
I think `Button_06` is pool heater on a RS-4 Combo.

**Reply by Bugman1400** [reply]
> So you have a dual panel, but no dual equipment. In that case, you should set it back to `RS-4 Combo` as the panels act differently and have different button configs.
> 
> You have not configured the pool heater on the buttons part of the config, that's why it's not showing up. I think `Button_06` is pool heater on a RS-4 Combo.

Pool Heater seems to have changed into an appropriate tile with a temp so, that's an improvement. How do I get rid of the Spa Water Temperature tile?
Also, how do I get my Waterfall Pump tile (which is a VSP) to show appropriately with an RPM reading?

![Screenshot from 2024-08-11 12-24-24](https://github.com/user-attachments/assets/646d3b05-9294-49e0-b782-712ec7dbb1f7)

**Comment by sfeakes** [comment]
This line `//"Temperature/Spa",`    should remove the tile.    So it looks like you have that correct. AqualinkD tells the browser to use very heavy caching, so if you changed that file and did not clear the cache in the browser, that could be the case.  So try clearing the cache in the browser, OR some browsers you can get around that by loading the file directly. `http://aqualink.ip/config.js`, load that page, hit refresh a few times, then try the main page again.

As for the VSP RPM, you have the wrong ID.  0x10 in the Pentair master ID, so that's the Jandy Control panel simulating a Pentair master device.  (run serial_logger with pump running to get the pump ID).  But even with wrong ID, you should have seen RPM from the control panel when the waterfall pump was turned on.  Were there any errors / warnings in the log?

**Comment by Bugman1400** [comment]
So, I seemed to have a Century V-Green 165 controller that needed programming before its recognized by my Aqualink RS4 controller. After some programming at the pump, it is now recognized as a Jandy ePump and has an address of 0x78.

![Screenshot from 2024-08-11 19-46-12](https://github.com/user-attachments/assets/579c61d2-43c6-43b3-8d32-ebc4b3801947)

So, I changed my aqualink.conf to:

button_02_label=Waterfall Pump
button_02_pumpID=0x78
button_02_pumpIndex=1

The Waterfall Pump still does not show as RPM. Also, I've noticed when I press the tile that my Aqualink RS4 controller seems to think that is the SPA Pump. Perhaps I have the wrong Button again.

**Comment by Bugman1400** [comment]
UPDATE: It seems there may be a bug with the T.2g firmware of the Aqualink RS4 Combo. Since my main filter pump in NOT a VSP and is single speed only, I set up my Waterfall VSP as an AUX Pump on address Pump1. However, it seems the VSP has to be set up as a Filtration Pump before being accepted. Then the VSP will work as intended and then you can set it back to an AUX Pump on Pump1. 

I also changed my aqualinkd.conf to enable read_RS485_ePump = yes and commented out #read_RS485_vsfPump:
read_RS485_ePump = yes
#read_RS485_vsfPump = yes

and the buttons to this:
button_05_label=Waterfall Pump
button_05_pumpID=0x78
button_05_pumpIndex=1

However, when I click on the Waterfall Pump tile the AUX3 turns ON. Am I missing something with how these buttons should be set? I want my Waterfall Pump tile just after my Filter Pump tile and then my Pool Light tile followed by my Pool Heater tile. I still can't seem to remove the Spa Water Temperature tile no matter what I do.

**Reply by sfeakes** [reply]
I’m not quite sure I understand the problem.  But let me explain how these work.
The button_xx_label is simply the label for the UI.  The xx corresponds to the button# in the Jandy protocol. This happens to be the allbutton keypad button #.   So if you have an allbutton keypad, that’s the easiest way to find the correct button,
If not is also the button# on the control panel. (Also the wiki has a table of this information, but RS4 is missing, but should be easy to see what that should be). So ultimately the button on/off state is dictated by the panel, the label is simply what you want to call it and has no bearing to the physical device.    Ie if the control panel has “Filter Pump” as button 01, and you label button 01 as “waterfall”, when you select “waterfall” in AqualinkD UI, the “Filter Pump” will turn on. 

The button_xx_pumpIndex & pumpID are simply what device/button to assign the watts/rpm/gpm information to, and has no other meaning.  IE you can assign this to any button and it won’t effect the way the control panel works. But for the UI, it will only display the watts/rpm/gpm if the button is turned on.

By default, the ordering of the buttons in the UI will be as the ordering on the panel.  Ie “Filter_Pump”, “Spa Mode”,  aux1, aux2, aux3 …….., “Pool Heater”, “Spa Heater”, “Solar Heater”.  If you want to change that, change the order of the list in `config.js`, but before you do that, you should get them all labeled and working correctly.

Remember to clear browser cache every time you edit `config.js`, and the default location is `/var/www/aqualinkd/config.js`, (some people edit the file in the download location, not the install location).  Again, point the browser to the file directly and make sure you get the right copy. (Listed in previous message of how to do that).

**Comment by Bugman1400** [comment]
I didn't know about the tile order of the buttons in the `config.js`, Not sure if that's listed in the Wiki or I just missed it, but it is very good info and should be added, if its not already. That will help me with some of my issues.
Since VSPs aren't related to a button, how are they handled and their tile placed in the preferred order?
Lastly, I still haven't figured out to get rid of the Spa Water Temperature tile. I've cleared the cache many times and have used other web browsers on other devices and it still remains.

Any info is appreciated.

**Reply by Bugman1400** [reply]
UPDATE: It seems I too suffered from the same affliction of others and was editing the wrong config.js file. Also, after some re-reading, I also found where you state that the order of Button list in the /etc/aqualinkd.conf dictate the order of the tiles on the UI.

I'm good now.

Thank you for this great app!
